cdap-runtime-ext-dataproc/src/main/java/co/cask/cdap/runtime/spi/provisioner/dataproc/DataProcConf.java

```diff
@@ -163,21 +206,39 @@ public static DataProcConf fromProvisionerContext(ProvisionerContext context) {
 
   /**
    * Create the conf from a property map while also performing validation.
+   *
+   * @throws IllegalArgumentException if it is an invalid config
    */
   public static DataProcConf fromProperties(Map<String, String> properties) {
     return create(properties, null);
   }
 
   private static DataProcConf create(Map<String, String> properties, @Nullable SSHPublicKey publicKey) {
     String accountKey = getString(properties, "accountKey");
+    if (accountKey == null) {
+      try {
+        getComputeEngineCredentials();
+      } catch (IOException e) {
+        throw new IllegalArgumentException(e.getMessage(), e);
+      }
+    }
     String projectId = getString(properties, "projectId");
+    if (projectId == null) {
+      projectId = getSystemProjectId();
+    }
 
     String region = getString(properties, "region");
     if (region == null) {
       region = "global";
     }
     String zone = getString(properties, "zone");
     String network = getString(properties, "network");
+    if (network == null) {
+      network = getSystemNetwork();
+    }
+    if (zone == null) {
+      getSystemZone();
+    }
 
     // a zone is always <region>-<letter>
     if (!"global".equals(region) && zone != null) {
```

Comment:
Not a part of this PR but region seem to be user input and can be in another case. If regions are case insensitive then we should ignore case here